bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2000,32 +2022,32 @@ private abstract class ImageFromImageDataProviderWrapper extends AbstractImagePr
 	void initImage() {
 		// As the init call configured some Image attributes (e.g. type)
 		// it must be called
-		newImageData(100);
+		newImageData(new ZoomContext(100));
 	}
 
 	@Override
-	ImageData newImageData(int zoom) {
+	ImageData newImageData(ZoomContext zoomContext) {
 		Function<Integer, ImageData> imageDataRetrieval = zoomToRetrieve -> {
-			ImageHandle handle = initializeHandleFromSource(zoomToRetrieve);
+			ImageHandle handle = initializeHandleFromSource(zoomContext);
 			ImageData data = handle.getImageData();
 			handle.destroy();
 			return data;
 		};
-		return cachedImageData.computeIfAbsent(zoom, imageDataRetrieval);
+		return cachedImageData.computeIfAbsent(zoomContext.targetZoom, imageDataRetrieval);
 	}
 
 	@Override
-	protected ImageHandle newImageHandle(int zoom) {
-		ImageData cachedData = cachedImageData.remove(zoom);
+	protected ImageHandle newImageHandle(ZoomContext zoomContext) {
+		ImageData cachedData = cachedImageData.remove(zoomContext.targetZoom);
 		if (cachedData != null) {
-			return newImageHandle(cachedData, zoom);
+			return newImageHandle(cachedData, zoomContext);
 		}
-		return initializeHandleFromSource(zoom);
+		return initializeHandleFromSource(zoomContext);
 	}
 
-	private ImageHandle initializeHandleFromSource(int zoom) {
-		ElementAtZoom<ImageData> imageDataAtZoom = loadImageData(zoom);
-		ImageData imageData = DPIUtil.scaleImageData(device, imageDataAtZoom.element(), zoom, imageDataAtZoom.zoom());
+	private ImageHandle initializeHandleFromSource(ZoomContext zoom) {
```

Comment:
NIT: preserve the naming standard
```suggestion
	private ImageHandle initializeHandleFromSource(ZoomContext zoomContext) {
```